test/python/transpiler/test_passmanager.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the passmanager logic"""
+
+import copy
+
+import numpy as np
+
+from qiskit import QuantumRegister, QuantumCircuit
+from qiskit.transpiler import PassManager
+from qiskit.transpiler import PropertySet
+from qiskit.compiler import transpile
+from qiskit.converters import circuit_to_dag
+from qiskit.transpiler.passes import Optimize1qGates, Unroller
+from qiskit.test.mock import FakeRueschlikon
+from qiskit.test import QiskitTestCase
+
+
+class TestPassManager(QiskitTestCase):
+    """Test Pass maanger logic."""
+
+    def test_callback(self):
+        """Test the callback parameter."""
+        qr = QuantumRegister(1, 'qr')
+        circuit = QuantumCircuit(qr, name='MyCircuit')
+        circuit.h(qr[0])
+        circuit.h(qr[0])
+        circuit.h(qr[0])
+        expected_start = QuantumCircuit(qr)
+        expected_start.u2(0, np.pi, qr[0])
+        expected_start.u2(0, np.pi, qr[0])
+        expected_start.u2(0, np.pi, qr[0])
+        expected_start_dag = circuit_to_dag(expected_start)
+
+        expected_end = QuantumCircuit(qr)
+        expected_end.u2(0, np.pi, qr[0])
+        expected_end_dag = circuit_to_dag(expected_end)
+
+        calls = []
+
+        def callback(**kwargs):
+            out_dict = kwargs
+            out_dict['dag'] = copy.deepcopy(kwargs['dag'])
+            calls.append(kwargs)
```

Comment:
Maybe `calls.append(out_dict)`?

Reply:
oops, yeah good catch, thanks. Fixed in: https://github.com/Qiskit/qiskit-terra/pull/2579/commits/2222bc18409afb21ea5b8d6e7e8e36a3e2d02866